fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -173,6 +175,7 @@ where
     pub async fn check_predicates_async<E>(
         checked: &Checked<Tx>,
         params: &CheckPredicateParams,
+        pool: VmPool,
```

Comment:
For asynchronous functions it makes sense to pass a pool. But maybe instead of the exact type, it is better to have a generic which implements some `VmPool` trait. In this case you don't need to implement a real `VmPool` in `fuel-vm` repository.

Reply:
We could. I don't see the point in making this any more complicated than it has to be, though. YAGNI and stuff.